src/persist/src/indexed/runtime.rs

```diff
@@ -102,40 +104,27 @@ where
     let indexed = Indexed::new(log, blob, maintainer, metrics.clone())?;
     let mut runtime = RuntimeImpl::new(config.clone(), indexed, rx, metrics.clone());
     let id = RuntimeId::new();
-    let runtime_pool = pool.clone();
-    let impl_handle = thread::Builder::new()
-        .name(format!("persist-runtime-{}", id.0))
-        .spawn(move || {
-            let pool_guard = runtime_pool.enter();
-            while runtime.work() {}
-            // Explictly drop the pool guard so the lifetime is obvious.
-            drop(pool_guard);
-        })?;
+    let impl_handle = tokio::spawn(async move { while runtime.work() {} });
 
     // Start up the ticker thread.
-    //
-    // TODO: Now that we have a runtime threaded here, we could use async stuff
-    // to do this.
     let ticker_tx = tx.clone();
-    let ticker_handle = thread::Builder::new()
-        .name(format!("persist-ticker"))
-        .spawn(move || {
-            // Try to keep worst case command response times to roughly `110% of
-            // min_step_interval` by ensuring there's a tick relatively shortly
-            // after a step becomes eligible. We could just as easily make this
-            // 2 if we decide 150% is okay.
-            let tick_interval = config.min_step_interval / 10;
-            loop {
-                thread::sleep(tick_interval);
-                match ticker_tx.send(Cmd::Tick) {
-                    Ok(_) => {}
-                    Err(_) => {
-                        // Runtime has shut down, we can stop ticking.
-                        return;
-                    }
+    let ticker_handle = tokio::spawn(async move {
+        // Try to keep worst case command response times to roughly `110% of
+        // min_step_interval` by ensuring there's a tick relatively shortly
+        // after a step becomes eligible. We could just as easily make this
+        // 2 if we decide 150% is okay.
+        let tick_interval = config.min_step_interval / 10;
+        loop {
+            thread::sleep(tick_interval);
```

Comment:
instead of `thread::sleep` you can use tokio's interval https://docs.rs/tokio/1.13.0/tokio/time/fn.interval.html